client/src/components/RecipeList/components/RecipeCard/RecipeCard.jsx

```diff
@@ -49,23 +58,28 @@ class RecipeCard extends React.Component {
       tags, id, isSaved = undefined,
     } = recipe;
     const viewMoreText = `${tags.length - 5} more`;
-    const { showModal, viewMoreTags, loadRecipeBookChange } = this.state;
+    const {
+      showModal, viewMoreTags, loadRecipeBookChange, loading,
+    } = this.state;
 
-    const openRecipe = () => {
+    const openRecipe = (e) => {
+      e.stopPropagation();
       window.open(link, 'noopener noreferrer');
       if (isLoggedIn) {
         viewRecipe(token, id);
       }
     };
 
-    const saveRecipe = () => {
+    const saveRecipe = (e) => {
+      e.stopPropagation();
       if (isLoggedIn) {
         this.setState({ loadRecipeBookChange: true });
         addRecipe(token, id);
       }
     };
 
-    const unsaveRecipe = () => {
+    const unsaveRecipe = (e) => {
+      e.stopPropagation();
```

Comment:
Why are all these necessary?

Reply:
That's the fix to the Modal opening when pressing the buttons at the bottom of the card. It didn't work for me without it.